adafruit_bus_device/i2c_device.py

```diff
@@ -164,8 +167,15 @@ def write_then_readinto(
     # pylint: enable-msg=too-many-arguments
 
     def __enter__(self):
-        while not self.i2c.try_lock():
-            pass
+
+        if self.timeout is not None:
+            lock_start_time = time.monotonic()
```

Comment:
Suggestion: use `time.monotonic_ns()` throughout as it'll avoid issues with floating point precision when the program runs for a long time.